modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/portlet/action/test/GetDataMVCResourceCommandTest.java

```diff
@@ -134,21 +140,133 @@ public void testGetContext() throws Exception {
 		String title = RandomTestUtil.randomString();
 
 		MockContextUtil.testWithMockContext(
-			new MockContextUtil.MockContext.Builder().analyticsReportsInfoItem(
-				MockAnalyticsReportsInfoItem.builder(
-				).authorName(
-					authorName
-				).authorProfileImage(
-					authorProfileImage
-				).publishDate(
-					publishDate
-				).title(
-					title
-				).build()
-			).build(),
+			new MockContextUtil.MockContext.Builder().
+				mockObjectAnalyticsReportsInfoItem(
+					MockObjectAnalyticsReportsInfoItem.builder(
+					).authorName(
+						authorName
+					).authorProfileImage(
+						authorProfileImage
+					).publishDate(
+						publishDate
+					).title(
+						title
+					).build()
+				).build(),
+			() -> {
+				MockLiferayResourceRequest mockLiferayResourceRequest =
+					_getMockLiferayResourceRequest(
+						new InfoItemReference(MockObject.class.getName(), 0L));
+
+				ServiceContext serviceContext = new ServiceContext();
+
+				serviceContext.setRequest(
+					mockLiferayResourceRequest.getHttpServletRequest());
+
+				ServiceContextThreadLocal.pushServiceContext(serviceContext);
+
+				MockLiferayResourceResponse mockLiferayResourceResponse =
+					new MockLiferayResourceResponse();
+
+				_mvcResourceCommand.serveResource(
+					mockLiferayResourceRequest, mockLiferayResourceResponse);
+
+				ByteArrayOutputStream byteArrayOutputStream =
+					(ByteArrayOutputStream)
+						mockLiferayResourceResponse.getPortletOutputStream();
+
+				JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
+					new String(byteArrayOutputStream.toByteArray()));
+
+				JSONObject contextJSONObject = jsonObject.getJSONObject(
+					"context");
+
+				JSONObject authorJSONObject = contextJSONObject.getJSONObject(
+					"author");
+
+				Assert.assertEquals(authorName, authorJSONObject.get("name"));
+				Assert.assertEquals(
+					authorProfileImage, authorJSONObject.get("url"));
+
+				Instant instant = publishDate.toInstant();
+
+				ZonedDateTime zonedDateTime = instant.atZone(
+					ZoneId.systemDefault());
+
+				LocalDate localDate = LocalDate.from(
+					DateTimeFormatter.ISO_DATE.parse(
+						contextJSONObject.getString("publishDate")));
+
+				Assert.assertEquals(zonedDateTime.toLocalDate(), localDate);
+
+				Assert.assertEquals(
+					title, contextJSONObject.getString("title"));
+
+				JSONArray jsonArray = contextJSONObject.getJSONArray(
+					"viewURLs");
+
+				Assert.assertEquals(
+					String.valueOf(jsonArray), jsonArray.length(), 1);
```

Comment:
The expected value should go first, right?

Reply:
yeah!!!